pyvista/plotting/composite_mapper.py

```diff
@@ -0,0 +1,769 @@
+        to our indexing to access the right block.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.block_attr[1].visible = False
+        >>> mapper.block_attr.reset_visibilities()
+        >>> pl.show()
+
+        """
+        self.RemoveBlockVisibilities()
+
+    def reset_pickability(self):
+        """Reset the pickability of all blocks.
+
+        Examples
+        --------
+        Make the cube of a multiblock dataset pickable and the sphere
+        unpickable, then reset it.
+
+        Note how the zero index is the entire multiblock, so we have to add 1
+        to our indexing to access the right block.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.block_attr[1].pickable = True
+        >>> mapper.block_attr[2].pickable = False
+        >>> mapper.block_attr.reset_pickability()
+
+        """
+        self.RemoveBlockPickabilities()
+
+    def reset_colors(self):
+        """Reset the color of all blocks.
+
+        Examples
+        --------
+        Set individual block colors and then reset it.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset, color='w')
+        >>> mapper.block_attr[1].color = 'r'
+        >>> mapper.block_attr[2].color = 'b'
+        >>> mapper.block_attr.reset_colors()
+        >>> pl.show()
+
+        """
+        self.RemoveBlockColors()
+
+    def reset_opacities(self):
+        """Reset the opacities of all blocks.
+
+        Examples
+        --------
+        Change the opacity of the second block of the dataset then reset all
+        opacities.
+
+        Note how the zero index is the entire multiblock, so we have to add 1
+        to our indexing to access the right block.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.block_attr[2].opacity = 0.5
+        >>> mapper.block_attr.reset_opacities()
+        >>> pl.show()
+
+        """
+        self.RemoveBlockOpacities()
+
+    def get_block(self, index):
+        """Return a block by its flat index.
+
+        Parameters
+        ----------
+        index : int
+            Flat index of the block to retrieve.
+
+        Returns
+        -------
+        pyvista.DataObject
+            PyVista dataobject.
+
+        Examples
+        --------
+        Note how the zero index is the entire multiblock. To access individual
+        multiblocks, add one to the index.
```

Comment:
I found this to be super confusing, and I'm sure everyone who comes across this via things like `multi.block_attr[1].color = 'blue'` will be confused too. Is there a VTK convention we are following here?

Reply:
Yes, I found it confusing at first and I've since clarified the docstring in aee6ffa58ee0f26d183b464f761ef056f110cae2.

See https://kitware.github.io/vtk-examples/site/Java/CompositeData/CompositePolyDataMapper/